platform/pulpcore/app/tasks/base.py

```diff
@@ -0,0 +1,56 @@
+from celery import shared_task
+from django.http import QueryDict
+
+from pulpcore.tasking.tasks import UserFacingTask
+from pulpcore.app.apps import get_plugin_config
+
+
+@shared_task(base=UserFacingTask)
+def general_update(instance_id, app_label, serializer_name, *args, **kwargs):
+    """
+    Update a model
+
+    The model instance is identified using the app_label, id, and serializer name. The serializer is
+    used to perform validation.
+
+    Args:
+        id (str): the id of the model
+        app_label (str): the Django app label of the plugin that provides the importer
+        serializer_name (str): name of the serializer class for the importer
+        data (dict): dictionary whose keys represent the fields of the importer that need to be
+            updated with the corresponding values.
```

Comment:
This is only perfectly true if partial=False (PATCH). 

Suggestion: dictionary whose keys represent the fields of the model and their corresponding values.